addons/mail/models/mail_thread.py

```diff
@@ -1967,6 +1967,40 @@ def message_post_with_template(self, template_id, **kwargs):
             composer.write(update_values)
         return composer.send_mail()
 
+    def _message_log(self, body='', subject=False, message_type='notification', **kwargs):
+        """ Shortcut allowing to post note on a document. It does not perform
+        any notification and pre-computes some values to have a short code
+        as optimized as possible. This method is private as it does not check
+        access rights and perform the message creation as sudo to speedup
+        the log process. This method should be called within methods where
+        access rights are already granted to avoid privilege escalation. """
+        if len(self.ids) > 1:
+            raise exceptions.Warning(_('Invalid record set: should be called as model (without records) or on single-record recordset'))
+
+        if kwargs.get('author_id'):
```

Comment:
Should probably `kwargs.pop('author_id')`, otherwise calling with e.g. `author_id=False` will set an email_from to the current user but will not set an author_id (because message_values is overwritten with all the kwargs).

Reply:
:+1: